2024-05-31-notes.md

```diff
@@ -0,0 +1,189 @@
+# 2024-05-31 - Jess and Orion notes
+
+Things we need to handle in version 2.0:
+
+- base properties for nodes
+- embed attached files
+
+- "extensions" through schemas
+
+- the schema doesn't have a single high-level version number. rather, each of the nodes and relations is defined by a schema with a version
+  - every node has a schema associated with it
+  - the base spec "includes" all of the node types in `@opencanvas/*`
+  - all of the schemas are currently embedded in file itself
+  - `x` and `y` are required when implementing a node type, optionally `z`
+
+Node Base class
+
+```ts
+type Node = {
+  id: string;
+  schema: string;
+  schema_version: string;
+  x: number;
+  y: number;
+  z?: number;
+};
+```
+
+## What node extensions are supported in the initial spec?
+
+- [x] text
+- [x] shape
+  - (all are modeled with x, y, w, h, rotation)
+    - rotation in degrees, + or - between 360 and -360, positive is clockwise
+  - rectangle
+  - ellipse (circle)
+  - triangle
+  - diamond
+  - hexagon
+  - capsule (oval in tldraw)
+    - a rectangle where the shortest two edges of the object are semi-circles
+  - parallelogram
+  - star
+  - cloud
+  - heart
+  - arrow
+  - checkbox
+- [x] arrows (relations are separate)
```

Comment:
what kind of arrows will be available via the spec? tldraw uses a bendable arc type for example, but other tools would benefit from other types like bezier cubic curves or svg like paths consisting of multiple segments

Reply:
Good question. We haven't fully defined the base `ocwg/arrow` type. Some discussion of arrows [here](https://github.com/orgs/ocwg/discussions/19#discussioncomment-9823419).